src/main/java/core/mate/academy/service/MachineServiceImpl.java

```diff
@@ -1,7 +1,47 @@
 package core.mate.academy.service;
 
+import core.mate.academy.model.Machine;
+import java.util.ArrayList;
+import java.util.List;
+
 /**
  * Your implementation of MachineService.
  */
-public class MachineServiceImpl {
+public class MachineServiceImpl implements MachineService<Machine> {
+
+    @Override
+    public List<Machine> getAll(Class<? extends Machine> type) {
+        List<? extends Machine> machineList;
+        switch (type.getSimpleName()) {
+            case "Bulldozer":
+                machineList = new BulldozerProducer().get();
+                break;
+
+            case "Excavator":
+                machineList = new ExcavatorProducer().get();
+                break;
+
+            case "Truck":
+                machineList = new TruckProducer().get();
+                break;
+
+            default:
+                machineList = new ArrayList<>();
+        }
+        return new ArrayList<>(machineList);
+    }
+
+    @Override
+    public void fill(List<? super Machine> machines, Machine value) {
+        for (int i = 0; i < machines.size(); i++) {
+            machines.set(i, value);
+        }
```

Comment:
The size of a list passed to `fill()` method should remain the same. Just replace the old values with the new ones, which is correctly implemented here.